### PR TITLE
Add LogSoftmax2d (SpatialLogSoftMax)

### PR DESCRIPTION
**Pull request by desimone**
See #551

### CODE REVIEW

File: torch/nn/modules/activation.py [hunk 1]
```diff
@@ -587,9 +614,29 @@ class LogSoftmax(Module):
...
+        - Input: :math:`(N, L, H, W)`
+        - Output: :math:`(N, L, H, W)`
+
+    Returns:
+        a Tensor of the same dimension and shape as the input with
+        values in the range [-inf, 0)
+    """
+
+    def forward(self, input):
+        return F.log_softmax(input)
```
Comment (apaszke) [inline torch/nn/modules/activation.py]: Are you sure regular log_softmax works with spatial inputs too?

**Review by apaszke — CHANGES_REQUESTED** [review]
You changed permissions on a lot of files (from 644 to 755 - added executable bit). Can you revert that please?

File: torch/nn/modules/activation.py (hunk 1 above)
Comment (apaszke) [inline torch/nn/modules/activation.py]: Ah, I see. Cool. Just fix the permissions please.

Comment (desimone) [inline torch/nn/modules/activation.py]: Sure thing!

**Comment by desimone** [comment]
@apaszke thanks for the review. I've reverted the permission changes.

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
@desimone the tests have failed

**Comment by apaszke** [comment]
Haha, yeah it know who should it listen to 😄

@pytorchbot test this please

**Comment by desimone** [comment]
I'm at a bit of a loss on this one. Maybe I over-reached.

```bash
FAIL: test_MultiLabelMarginLoss (__main__.TestNN)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_nn.py", line 1157, in <lambda>
    setattr(TestNN, test_name, lambda self,test=test: test(self))
  File "/tmp/builder/pytorch/test/common_nn.py", line 595, in __call__
    test_case.check_criterion_jacobian(module, input, self.target)
  File "/tmp/builder/pytorch/test/common_nn.py", line 444, in check_criterion_jacobian
    PRECISION
AssertionError: 2.0637550102259317e-05 not less than or equal to 1e-05

----------------------------------------------------------------------
Ran 265 tests in 26.546s

FAILED (failures=1, skipped=4)Build step 'Execute shell' marked build as failure
```

**Comment by desimone** [comment]
Any thoughts on why this would cause the `MultiLabelMarginLoss` test to fail `check_criterion_jacobian`? The connection is non-obvious to me.

**Comment by soumith** [comment]
@desimone afai can see, this is not a failure from you. The ordering of the tests changed and we found that we need to bump the tolerance on that test. :)

**Comment by apaszke** [comment]
Actually we've discussed this, and decided that since the code is exactly the same for `LogSoftmax` and `LogSoftmax2d`, we'd rather keep only the regular one. Can you just limit your changes to `test/test_nn.py`, but use `LogSoftmax` in that test case? Thanks!

**Comment by desimone** [comment]
Makes sense given LogSoftmax (seems to be) a superset of `LogSoftMax` and `SpatialLogSoftMax`. Let me know if this is what you had in mind.

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
Thanks!

**Pull request merged**